libbpf-tools/task_detector.bpf.c

```diff
@@ -0,0 +1,264 @@
+// SPDX-Licence-Identifier: GPL-2.0
+// Copyright (c) 2020 Alibaba Cloud
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include "task_detector.h"
+
+#define DEBUG_ON 0
+#define TASK_RUNNING 0x0000
+#define TASK_REPORT_MAX 0x0100
+
+struct{
+	__uint(type, BPF_MAP_TYPE_ARRAY);
+	__uint(max_entries, 1);
+	__type(key, int);
+	__type(value, struct user_info);
+} user_info_maps SEC(".maps");
+
+struct{
+	__uint(type, BPF_MAP_TYPE_ARRAY);

```

Comment:
this can also be just a global variable array:

```
struct trace_info trace_infos[NR_ENTRY_MAX];
```

From user-space, iterating it will be also nice and simple:

```
for (i = 0; i < NR_ENTRY_MAX; i++) {
    struct trace_info *ti = &skel->bss->trace_infos[i];
   ...
}
```